storage/dataflux/fast_list.go

```diff
@@ -131,31 +131,53 @@ func (c *Lister) NextBatch(ctx context.Context) ([]*storage.ObjectAttrs, error)
 	var results []*storage.ObjectAttrs
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
+
+	wsCompletedfirst := false
+	seqCompletedfirst := false
+	var wsObjects []*storage.ObjectAttrs
+	var seqObjects []*storage.ObjectAttrs
+	var nextToken string
+
 	// Errgroup takes care of running both methods in parallel. As soon as one of
 	// the method is complete, the running method also stops.
 	g, childCtx := errgroup.WithContext(ctx)
 
 	// To start listing method is Open and runs both worksteal and sequential listing
 	// in parallel. The method which completes first is used for all subsequent runs.
 
-	// TODO: Run worksteal listing when method is Open or WorkSteal.
+	// Run worksteal listing when method is Open or WorkSteal.
+	if c.method != sequential {
+
+		g.Go(func() error {
+			objects, err := c.workstealListing(childCtx)
+			if err != nil {
+				countError++
```

Comment:
Won't this be a race condition when c.method is open and both sequential and worksteal are running at the same time? Can we hit this case in a emulator test?

Reply:
Added unit test to catch this in the race conditions.